### PR TITLE
Add a build.sh script

### PR DESCRIPTION
**Pull request by mestery**
This commit moves the build logic out of the Dockerfile and into
a script, thus allowing a user to run the build without having
to copy and paste from the Dockerfile.

Signed-off-by: Kyle Mestery <mestery@mestery.com>

### CODE REVIEW

File: scripts/build.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+#
+# This script builds the networkservicemesh
+#
+
```
Comment (edwarnicke) [inline scripts/build.sh]: I think we want to retain:

test -z $(go fmt ./...) || (echo "Run go fmt ./... and recommit your code";exit 1)

To do the format checking.

File: scripts/build.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+go get -u github.com/golang/protobuf/protoc-gen-go
+go generate ./...
+go install ./...
+go test ./...
```
Comment (fkautz) [inline scripts/build.sh]: Can you add -race to this?

Race doesn't provide false positives, so it should be safe to add.

File: scripts/build.sh (hunk 1 above)
Comment (mestery) [reply scripts/build.sh]: I had thought I left that in, but I will add it back, as it does appear to be gone.

File: scripts/build.sh (hunk 2 above)
Comment (mestery) [reply scripts/build.sh]: Will do!

File: scripts/build.sh (hunk 1 above)
Comment (mestery) [reply scripts/build.sh]: Actually, the format check remains in the Dockerfile, see line 5 above. Let me know if you'd prefer it moved here.

Comment (edwarnicke) [reply scripts/build.sh]: Ah... I missed that.  I could see either way, so long as we are deliberate about it :)

File: scripts/build.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+#
+# This script builds the networkservicemesh
+#
+
+go get -u github.com/golang/protobuf/protoc-gen-go
+go generate ./...
+go install ./...
+go test -race ./...
```
Comment (mestery) [inline scripts/build.sh]: So it turns out using "go test -race" with alpine linux is a non-starter [1]. I am going to remove this for now.

[1] https://github.com/golang/go/issues/14481

File: scripts/build.sh (hunk 2 above)
Comment (mestery) [reply scripts/build.sh]: @fkautz See here as to why "go test -race" won't work with alpine linux:

https://github.com/golang/go/issues/14481

File: scripts/build.sh (hunk 1 above)
Comment (mestery) [reply scripts/build.sh]: If that's the case, let me move it, as it may make more sense to have it in one place.

**Pull request merged**